packages/orm/lib/orm.js

```diff
@@ -0,0 +1,235 @@
+const chunkArray = (array, chunkSize) => {
+  if (chunkSize <= 0) chunkSize = 1;
+
+  return array.reduce((accum, item, index) => {
+    const chunkIndex = Math.floor(index / chunkSize);
+
+    if (!accum[chunkIndex]) {
+      accum[chunkIndex] = [];
+    }
+
+    accum[chunkIndex].push(item);
+
+    return accum;
+  }, []);
+};
+
+const runQuery = async ({
+  query,
+  variables,
+  keystone,
+  schemaName = 'public',
+  extraContext = {},
+}) => {
+  const { data, errors } = await keystone.executeGraphQL({
+    context: {
+      ...keystone.createContext({ schemaName, authentication: {}, skipAccessControl: true }),
+      ...extraContext,
+    },
+    query,
+    variables,
+  });
+  if (errors) throw errors[0];
+  return data;
+};
+
+const runPaginatedQuery = async ({
+  pageSize = 1,
+  variables = {},
+  verbose = false,
+  where = {},
+  ...rest
+}) => {
+  let skip = 0;
+  let latestResult;
+  const allItems = [];
+
+  if (verbose) {
+    console.log(`Started paginated query`);
+    console.log({ pageSize });
+  }
+
+  do {
+    if (verbose) console.log({ skip });
+
+    const response = await runQuery({
+      ...rest,
+      variables: { ...variables, first: pageSize, skip, where },
+    });
+
+    latestResult = response[Object.keys(response || {})[0]];
+
+    allItems.push(...latestResult);
+
+    skip += pageSize;
+  } while (latestResult.length);
+
+  if (verbose) {
+    console.log(`Finished paginated query`);
+    console.log(`${allItems.length} items found`);
+  }
+
+  return allItems;
+};
+
+const runChunkedMutation = async ({ gqlName, pageSize = 500, items, ...rest }) => {
```

Comment:
Let's name this paginate query and move the contents of `chunkedArray` inline here as it's the only place it's called

Reply:
Alternatively we could just call this `runMutation`

Reply:
Hah just realised we have `runPaginatedQuery` so `runPaginatedMutation` makes sense